### PR TITLE
Straylight2

### PR DESCRIPTION
**Pull request by jemorrison**
added straylight removal by using Modified Shepard Method to  find the distances of the pixels in the slice gaps. 
This method is set to be used by default, but the other straylight method was kept and can be used
by --method=Nearest. We can probably remove the old method (nearest) at a later date after INS has tested this method. For now I left it.

### CODE REVIEW

**Comment by jemorrison** [comment]
I removed 'channel' from IFUCube model schema. This will all be handled by model_blender

File: jwst/straylight/straylight.py [hunk 1]
```diff
@@ -214,12 +185,139 @@ def correct_MRS(input_model, straylight_model):
...
+    #plt.show()
+
+    # The regions mask has values of 0 and nonzero, values of 0 present pixel
+    # in a slice gap and non zero values are science pixels.  The straylight task uses data
+    # in-between the slices (also called slice gaps) of the MRS data to correct 
+    # the science data in the slices. 
+
+    # Create output as a copy of the input science data model
+    output = input_model.copy() # this is used in algorithm to
+    output_data = input_model.data.copy()
```
Comment (hbushouse) [inline jwst/straylight/straylight.py]: Why is `output_data` needed as a separate array? Can't you just use `output.data`? They both contain the same array.

File: jwst/straylight/straylight_step.py [hunk 2]
```diff
@@ -3,45 +3,73 @@
...
-
+from ..datamodels import RegionsModel
 
 class StraylightStep (Step):
     """
     StraylightStep: Performs straylight correction image using a Mask file.
     """
-    reference_file_types = ['straymask']
+
+    spec = """
```
Comment (hbushouse) [inline jwst/straylight/straylight_step.py]: These new arguments also need to be added to the default `straylight.cfg` file in the pipeline directory.

**Comment by hbushouse** [comment]
This PR also contains updates to cube_build from #1737

File: jwst/cube_build/cube_build_wcs_util.py [hunk 3]
```diff
@@ -582,6 +648,82 @@ def set_geometryAB(self, footprint):
 
 
 
+#_______________________________________________________________________
+# from a set of ra values find the average ra.
+# This is tricky because of the convergence of hour angles
+# this method is taken from the SPITZER SSC MOSAICER tool
+def average_ra(ravalues):
+
```
Comment (crawfordsm) [inline jwst/cube_build/cube_build_wcs_util.py]: I was just taking a quick look through this, but did you consider using [astropy.units.Angle](http://docs.astropy.org/en/stable/coordinates/angles.html) and [astropy.stats.circmean](http://docs.astropy.org/en/stable/stats/circ.html)?

Comment (jemorrison) [reply jwst/cube_build/cube_build_wcs_util.py]: I was trying to find something in astro py that did something like this. I thought it must exist but
I could not find it. I will look at those two routines in astropy.

**Comment by jemorrison** [comment]
Now using astropy.stats.circmean to find the average right ascension

File: jwst/straylight/straylight_step.py (hunk 2 above)
Comment (jemorrison) [reply jwst/straylight/straylight_step.py]: Do we put all the default values in the cfg files  ?
If so then I need to update the cube_build on too.

Comment (hbushouse) [reply jwst/straylight/straylight_step.py]: Yes. See, for example, https://github.com/STScI-JWST/jwst/blob/master/jwst/pipeline/skymatch.cfg

**Comment by hbushouse** [comment]
Given that the straylight algorithm has been replaced I assume this means the documentation needs to be replaced/updated too. I don't see documentation updates in this PR. That needs to be done yet.

**Pull request merged**